examples/full_pipeline.rs

```diff
@@ -49,39 +49,21 @@ impl SquareOperator {
     }
 }
 
-struct SquareOperatorState {
-    current_timestamp: Timestamp,
-}
-
-impl SquareOperatorState {
-    fn new() -> Self {
-        Self {
-            current_timestamp: Timestamp::Bottom,
-        }
-    }
-}
-
-impl StateT for SquareOperatorState {
-    fn commit(&mut self, timestamp: &Timestamp) {
-        self.current_timestamp = timestamp.clone();
-    }
-
-    fn get_last_committed_timestamp(&self) -> Timestamp {
-        self.current_timestamp.clone()
-    }
-}
-
-impl OneInOneOut<SquareOperatorState, usize, usize> for SquareOperator {
-    fn setup(&mut self, ctx: &mut SetupContext<SquareOperatorState>) {
+impl OneInOneOut<TimeVersionedState<()>, usize, usize> for SquareOperator {
```

Comment:
Why is this time-versioning a tuple? Wouldn't it make more sense to time-version a number, or provide `()` as the state?